stistools/defringe/prepspec.py

```diff
@@ -61,8 +67,64 @@ def prepspec(inspec, outroot, darkfile=None, pixelflat=None, initguess='min'):
     # If G750M:
     #    perform only {WAVECORR, HELCORR, X2DCORR} via stistools.x2d.x2d()
     #    return SX2 file
-
-    raise NotImplementedError()
+    science_data = os.path.abspath(expandFileName(inspec))
+    sci_root = re.split('\.fits.*', os.path.basename(science_data), flags=re.IGNORECASE)[0].rsplit('_', 1)[0]
+    opt_elem = fits.getval(science_data, 'OPT_ELEM').strip().upper()
+    if (fits.getval(science_data, ext=0, keyword='DETECTOR').strip().upper() != 'CCD') or \
+            (fits.getval(science_data, ext=0, keyword='INSTRUME').strip() != 'STIS'):
+        raise Exception('prepspec:  Intended for use on STIS/CCD data!')
+
+
+    # Make sure the necessary header keywords are set to PERFORM
+    fits.setval(science_data, 'BLEVCORR', value='PERFORM')  # overscan correction
```

Comment:
Testing if the keyword flag is already complete allows you to set only the ones needed.  Try something like:

```python
with fits.open(science_data, 'update') as f:
    for keyword in ['BLEVCORR', 'CRCORR', 'BIASCORR', 'DARKCORR', 'FLATCORR']:
        if not f[0].header[keyword].upper().startswith('COMPLETE'):
            f[0].header[keyword] = 'PERFORM'
```